OpenDreamShared/Dream/DreamPath.cs

```diff
@@ -214,25 +213,32 @@ public enum PathType {
 
         public static bool operator !=(DreamPath lhs, DreamPath rhs) => !(lhs == rhs);
 
-        private void Normalize()
+        private void Normalize(bool canHaveEmptyEntries)
         {
-            var writeIdx = 0;
-            for (var i = 0; i < _elements.Length; i++)
+            if (canHaveEmptyEntries && _elements.Contains(""))
+            {
+                // Slow path :(
+                _elements = _elements.Where(el => !string.IsNullOrEmpty(el)).ToArray();
```

Comment:
Please avoid using LINQ

Reply:
This slow path executes very rarely, I don't think it's a big performance concern.